docs/design/kangcheolung-#240-collection-list-pagination.md

```
@@ -0,0 +1,311 @@
+
+```java
+// 일부러 되돌린 코드
+return collectionRepository.findReadableChildren(collectionId, userId)
+        .stream()
+        .filter(child -> permissionQueryService.canReadCollection(userId, child)) // N+1 재현
+        .map(collectionConverter::toResponse)
+        .toList();
+```
+
+결과:
+```
```

markdownlint-cli2 (0.23.2)

[warning] Fenced code blocks should have a language specified

(MD040, fenced-code-language)